### DISCUSSION TITLE
Just got my yoga 7x, time to install nix

### DISCUSSION THREAD

**Discussion by eureka-cpu** [kuruczgy/x1e-nixos-config#53] [General]
Howdy, just wanted to say really stoked on this repo. I'm installing tonight, and happy to test things out along the way. I'm not super knowledgeable on getting NixOS to work on new hardware, but I am experienced in writing nix and I'm happy to take on tasks so long as there is direction. Cheers!

**Comment by eureka-cpu** [comment]
I'm unfortunately stuck on the efi part in the readme. I seem to have a frozen screen while it's loading in :/ any tips?

**Reply by eureka-cpu** [reply]
Ok it worked after a few tries, I guess it just gets stuck on occasion.

**Comment by eureka-cpu** [comment]
Ok, I was able to boot into NixOS, and I see that the sway DE was enabled but it doesn't seem to boot in with sway, is that expected?

**Reply by eureka-cpu** [reply]
It just didn't start automatically, but it did work after starting manually. Also can confirm the WiFi works but wpa_cli fails, so I just used nmtui. 

Man this is snappy, windows on this thing is clunky in comparison.

**Reply by kuruczgy** [reply]
Both valid points from a documentation improvement perspective:
- There is no login manager installed (in fact sway doesn't support one officially). Sway is just included as a convenience so you have something graphical to test. I could describe this at the end of the README.
- About WiFi, you are the second person trying to use wpa_cli in the installed system, so I should definitely mention in the README that that is only meant for the installer, the installed system uses NetworkManager, and you have to use nmtui (as you correctly concluded).

**Comment by eureka-cpu** [comment]
I think I know how to fix it, but I'm running into a kernel version mismatch when rebuilding. It's getting late tonight so I'll have to pick up tomorrow, probably the problem is that the iso has a version of nixpkgs which comes pre-patched with the overlay, but I'm not certain. Any tips are welcome, I usually haven't had to care too much about which kernel version I'm on :p 

```
error: builder for '/nix/store/gcqafnlvl1p0vjpax8kc5k294klyrxql-linux-6.12.0-modules.drv' failed with exit code 1;
       last 3 log lines:
       > created 5 symlinks in user environment
       > inconsistent kernel versions: 6.12.0
       > 6.6.63
       For full logs, run 'nix log /nix/store/gcqafnlvl1p0vjpax8kc5k294klyrxql-linux-6.12.0-modules.drv'.
```

**Reply by kuruczgy** [reply]
Not sure what could be causing that error.

We are setting `nixpkgs.overlays` in the module, so unless you are doing something weird like setting `specialArgs.pkgs` (someone recently had an issue like that: https://github.com/kuruczgy/x1e-nixos-config/issues/50#issuecomment-2502048357), the overlays should apply fine.

I would try temporarily setting your nixpkgs version to the exact one found in this repo's lockfile, to rule nixpkgs version mismatch out as an issue.

Other than that, IMO the most likely explanation is some conflicting option in your config. (If you have it publicly available maybe I could take a look later.)

**Reply by eureka-cpu** [reply]
Here's what I got: https://github.com/eureka-cpu/dotfiles/tree/eureka-cpu/add-xie

**Reply by kuruczgy** [reply]
If I had to guess this is your issue: https://github.com/eureka-cpu/dotfiles/blob/8e1e1e4026fbac77b854490288877efb74341316/users/eureka/nixos/configuration.nix#L15

I think `pkgs.linuxPackages` is `6.6.63` in your error message above. First you should just try removing any v4l2loopback stuff, see if that fixes the issue. Afterwards, if you really want v4l2loopback, maybe something like `boot.extraModulePackages = [ config.boot.kernelPackages.v4l2loopback ];` could work? (If it compiles...)

**Reply by eureka-cpu** [reply]
yeeep that was it, thanks haha

**Comment by eureka-cpu** [comment]
I see that the battery status in the readme is working but on my install I just get "not found" in the default sway bar. How can I test that this is working properly?

**Reply by kuruczgy** [reply]
`cat /sys/class/power_supply/qcom-battmgr-bat/energy_now` to see the raw battery level as reported by the kernel.

If that's broken, check the remoteproc statuses with `cat /sys/class/remoteproc/*/state`, they should both report `running`.

**Reply by eureka-cpu** [reply]
Ah, that does work thanks. I think maybe the bar is just assuming that a `BAT0` is supposed to be somewhere, and it isn't?

**Comment by eureka-cpu** [comment]
@kuruczgy if you have any instructions I can follow to help get the PCIe and power profiles working I'm happy to work on those, I'm just not sure where to start 🙂

**Reply by kuruczgy** [reply]
If you mean PCIe over USB-C, I thought about it a bit a while ago, the only interesting use-case I could think of is eGPU, and I did not deem it to be worth my time:
- eGPU enclosures are very overpriced IMO, so not worth getting one "just to play around"
- even if we got it working perfectly, it would be limited to 4x PCIe lanes, which is kinda slow for modern GPUs

Regarding power profiles, I think it would first make more sense to fix thermal throttling (right now there is none, so the laptop can overheat.) Either way, you would probably wanna start by asking around in #aarch64-laptops: https://github.com/aarch64-laptops/build#contact-us

**Reply by eureka-cpu** [reply]
Ah I see. Really I just want to be able to use USBC and thunderbolt accessories for my external display, mouse, keyboard and audio.

**Reply by kuruczgy** [reply]
Ah yeah, I have heard that some thunderbolt docks expose e.g. a downstream UBS controller over PCIe for their USB ports. So yeah, we might indeed need PCIe (altmode/tunneling? not sure which) for that.

But FWIW all those things (display, input peripherals, audio) already work with the dongle that came with the laptop.

**Reply by eureka-cpu** [reply]
My laptop did not come with a dongle sadly </3 Just the laptop and charger.

**Reply by eureka-cpu** [reply]
Would you mind sharing the part number for the dongle that came with it?

**Reply by kuruczgy** [reply]
It has these:
```
Model PN: LOHC474
OPT PN: GX91G98509
```

Maybe they only include a dongle in the EU? (But if I were you I would also double check the box of the laptop to make sure it did not get accidentally left in there :smile: )
